src/par_iter/mod.rs

```diff
@@ -609,18 +609,19 @@ pub trait ParallelIterator: Sized {
     fn drive_unindexed<C>(self, consumer: C) -> C::Result
         where C: UnindexedConsumer<Self::Item>;
 
+    /// Internal method used to fake specialization for indexed collect.
+    #[doc(hidden)]
+    fn opt_len(&mut self) -> Option<usize> { None }
```

Comment:
I made it hidden and internal so we could feel free to remove it later, when we're able to accomplish the same thing with specialization.  I'm not sure it's much use outside of that purpose.  Maybe we can write it as `size_hint` if you want to mirror `Iterator`, but I don't think we *need* that anywhere.

Anyway, your expanded comment is fine with me -- I'll take it verbatim if you don't mind.